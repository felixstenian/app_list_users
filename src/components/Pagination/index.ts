export { default } from './Pagination'
export type { PaginationProps } from './Pagination'
export { default as PaginationItem } from './PaginationItem'
