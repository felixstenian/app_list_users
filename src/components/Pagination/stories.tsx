import { Meta, Story } from '@storybook/react'
import { useState } from 'react'

import Pagination, { PaginationProps } from '.'

export default {
  title: 'Pagination',
  component: Pagination,
  args: {}
} as Meta

export const Default: Story<PaginationProps> = (args) => {
  const [currentPage, setCurrentePage] = useState(1)
  return (
    <Pagination
      {...args}
      currentPage={currentPage}
      onPageChange={setCurrentePage}
    />
  )
}
