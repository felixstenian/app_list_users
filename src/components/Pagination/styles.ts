import styled from 'styled-components'

export const Button = styled.button`
  width: 2rem;
  border: 0;
  background: transparent;
  cursor: pointer;

  :disabled {
    cursor: auto;
    width: 3rem;
  }
`
