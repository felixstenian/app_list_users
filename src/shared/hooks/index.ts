export { default as useKeyPress } from './useKeyPress'
export { default as useOnClickOutside } from './useOnClickOutside'
export { default as useSnackBar } from './useSnackBar'
