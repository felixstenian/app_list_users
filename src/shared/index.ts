export * as types from './types'
export * as helpers from './helpers'
export * as utils from './utils'
