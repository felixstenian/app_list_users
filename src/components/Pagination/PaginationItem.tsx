import { Text } from 'components'
import * as Style from './styles'

interface PaginationItemProps {
  number: number
  isCurrent?: boolean
  onPageChange: (page: number) => void
  value?: React.ReactNode | string
}

const PaginationItem = ({
  isCurrent = false,
  number,
  onPageChange,
  value
}: PaginationItemProps) => {
  if (isCurrent) {
    return (
      <Style.Button disabled>
        <Text variant="smallTitle" color="black">
          {number}
        </Text>
      </Style.Button>
    )
  }

  return (
    <Style.Button data-testid="next-page" onClick={() => onPageChange(number)}>
      {value || number}
    </Style.Button>
  )
}

export default PaginationItem
