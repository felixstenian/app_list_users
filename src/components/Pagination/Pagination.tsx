import { Row } from 'components'
import { GrNext, GrPrevious } from 'react-icons/gr'
import PaginationItem from './PaginationItem'

export interface PaginationProps {
  currentPage?: number
  onPageChange: (page: number) => void
}

const Pagination = ({ currentPage = 1, onPageChange }: PaginationProps) => {
  const previousPages = currentPage - 1
  const nextsPages = currentPage + 1

  return (
    <Row
      py={10}
      justifyContent="center"
      alignContent="center"
      data-testid="pagination"
    >
      {previousPages > 0 && (
        <PaginationItem
          onPageChange={onPageChange}
          key={previousPages}
          number={previousPages}
          value={<GrPrevious />}
        />
      )}

      <PaginationItem
        onPageChange={onPageChange}
        number={currentPage}
        isCurrent
      />

      {nextsPages > 0 && (
        <PaginationItem
          onPageChange={onPageChange}
          key={nextsPages}
          number={nextsPages}
          value={<GrNext />}
        />
      )}
    </Row>
  )
}

export default Pagination
