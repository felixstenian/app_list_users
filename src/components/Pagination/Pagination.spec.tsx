import { fireEvent, screen } from '@testing-library/react'

import { renderWithTheme } from 'shared/utils'
import Pagination from '.'

describe('<Pagination />', () => {
  it('1. should render the pagination component with the correct props', () => {
    const onPageChange = jest.fn()
    const { container } = renderWithTheme(
      <Pagination currentPage={1} onPageChange={onPageChange} />
    )

    expect(screen.getByTestId('pagination')).toBeInTheDocument()
    expect(screen.getByText('1')).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
  it('2. should call the onPageChange function when a page is clicked', () => {
    const onPageChange = jest.fn()
    renderWithTheme(<Pagination currentPage={1} onPageChange={onPageChange} />)
    const nextPage = screen.getByTestId('next-page')
    fireEvent.click(nextPage)
    expect(onPageChange).toHaveBeenCalledWith(2)
  })
})
