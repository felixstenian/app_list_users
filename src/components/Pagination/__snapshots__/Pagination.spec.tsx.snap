// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pagination /> 1. should render the pagination component with the correct props 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding-top: 10px;
  padding-bottom: 10px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
}

.c2 {
  color: black;
  font-family: Poppins,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Open Sans','Helvetica Neue',sans-serif;
  font-size: 1.8rem;
}

.c1 {
  width: 2rem;
  border: 0;
  background: transparent;
  cursor: pointer;
}

.c1:disabled {
  cursor: auto;
  width: 3rem;
}

<div
  class="c0"
  data-testid="pagination"
>
  <button
    class="c1"
    disabled=""
  >
    <h3
      class="c2"
      color="black"
    >
      1
    </h3>
  </button>
  <button
    class="c1"
    data-testid="next-page"
  >
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <polyline
        fill="none"
        points="7 2 17 12 7 22"
        stroke="#000"
        stroke-width="2"
      />
    </svg>
  </button>
</div>
`;
